dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/DatasetServiceIT.java

```diff
@@ -0,0 +1,65 @@
+package edu.harvard.iq.dataverse.dataset;
+
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.GenericDao;
+import edu.harvard.iq.dataverse.arquillian.arquillianexamples.WebappArquillianDeployment;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.inject.Inject;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class DatasetServiceIT extends WebappArquillianDeployment {
+
+    @Inject
+    private DatasetService datasetService;
+
+    @Inject
+    private DataverseSession dataverseSession;
+
+    @Inject
+    private AuthenticationServiceBean authenticationServiceBean;
+
+    @Inject
+    private GenericDao genericDao;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void removeDatasetThumbnail() {
+        //given
+        Dataset datasetWithFiles = genericDao.find(52, Dataset.class);
+        datasetWithFiles.setThumbnailFile(datasetWithFiles.getFiles().get(0));
+
+        //when
+        datasetService.removeDatasetThumbnail(datasetWithFiles);
+        Dataset updatedDataset = genericDao.find(52, Dataset.class);
+
+        //then
+        Assert.assertNull(updatedDataset.getThumbnailFile());
```

Comment:
Test kind of doesn't make sense. Even if we remove line `datasetService.removeDatasetThumbnail(datasetWithFiles);` it would pass

Reply:
What do you mean, if i remove this method call it won't pass :P.

Reply:
Ok. But can we explicitly merge dataset before `when` - that way we don't care if auto save will occur or not

Reply:
I had some thoughts about it earlier, why do we use JPA if we don't utilize one of it's primary feature? You can even say it can be misleading for people who don't know it, because you say in the code that em.merge actually saves the changes. And it is not true because it is already in cache, so someone can assume that since there is a merge, if I remove it it won't merge changes right? And someone could be surprised that it still works. (This answer is more general than only for this test :P).

Reply:
I just think this is more error prone. We could imagine the scenario:
1) Let say we needed to make some other test in class non transactional
2) We remove transactional annotation on class
3) And here is tricky part - Most people would run tests and see that they are passing (I would do that) and think ok - I didn't broke anything. We force on developer to check every test and understand it again.
4) Now even if something was wrong with method then nothing would get reported.